demos/graph-classification/dgcnn-graph-classification.ipynb

```diff
@@ -0,0 +1,761 @@
+{
```

Comment:
<p>It looks like this doesn't kick in; should we increase the epochs further to train to "completion"?</p><p> </p><p>In terms of the length of time to train, it might be noticeably reduced by https://github.com/stellargraph/stellargraph/pull/1381 which I just merged, and, unlike the other graph classification notebook, there's lots of progress updates for the user to tell that something is happening here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1319/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Reply:
<p>Early stopping doesn't have to kick in if the quantity monitored does not require it. It is a failsafe more than anything to prevent overfitting. Different runs will produce different results with regards to convergence depending on the data split and initial weight values so for a different run maybe early stopping will be used. </p><p> </p><p>It might be best that we follow the training scheme in the paper and train for 100 epochs with no early stopping and just report the performance for the model at the 100th epoch.</p><p> </p><p>So I guess we have to options (I'm sure there are more but keeping it simple):</p><ol><li>Remove early stopping run for 100 epochs as in the paper</li><li>Increase the number of epochs</li></ol><p> </p><p>Also, for all demos we set <code>verbosity=0</code>  for the final version. In the current version I have it set to 1 for illustration. Shall I leave it as 1 so the user received feedback during training or set it to 0 like the other notebooks?</p><p> </p>

Reply:
Let me rephrase: it looks like the loss is still decreasing (and accuracy still increasing). My understanding is that early stopping is a low-effort way to avoid having to fine tune the number of epochs: just choose a large number, and let early stopping detect when the model looks like it's learnt as much as it can under the given configuration.

However, following the paper without early stopping (i.e. 1) is good too.

> Also, for all demos we set verbosity=0 for the final version. In the current version I have it set to 1 for illustration. Shall I leave it as 1 so the user received feedback during training or set it to 0 like the other notebooks?

I don't think this is quite the case any more:

```
$ cd demos
$ git grep verbose=0 | wc -l
      22
$ git grep verbose=1 | wc -l
      13
$ git grep verbose=2 | wc -l
      21
```

From quick glance over the actual results, I think all of the `verbose=0` are in code, all but one of the `verbose=1` results, and similarly all but one of the `verbose=2` results. This leaves 22 `verbose=0` calls vs. 32 `verbose>0 ` calls.

Given this, and given it's a much better user experience (although mildly unfortunate that it's 100 lines of automated output), I suggest leaving it.

Reply:
Ok, so I'll just set `epochs` to 150 and leave early stopping with patience at 25 so it neither runs too long and still have a chance to find a good model. Would that be ok or do you prefer an even larger value for epochs?

Reply:
Ok then, I changed it to follow the paper removing early stopping and training for exactly 100 epochs.

I think I have now addressed all your suggestions so let me know if anything else needs to be updated.